sonar-scanner-engine/src/main/java/org/sonar/scanner/report/ContextPropertiesPublisher.java

```diff
@@ -20,31 +20,45 @@
 package org.sonar.scanner.report;
 
 import com.google.common.base.Function;
+import com.google.common.collect.Iterables;
 import java.util.Map;
+import javax.annotation.Nonnull;
+import org.sonar.api.config.Settings;
 import org.sonar.scanner.protocol.output.ScannerReport;
 import org.sonar.scanner.protocol.output.ScannerReportWriter;
 import org.sonar.scanner.repository.ContextPropertiesCache;
 
 import static com.google.common.collect.FluentIterable.from;
 
 public class ContextPropertiesPublisher implements ReportPublisherStep {
+
   private final ContextPropertiesCache cache;
+  private final Settings settings;
 
-  public ContextPropertiesPublisher(ContextPropertiesCache cache) {
+  public ContextPropertiesPublisher(ContextPropertiesCache cache, Settings settings) {
     this.cache = cache;
+    this.settings = settings;
   }
 
   @Override
   public void publish(ScannerReportWriter writer) {
-    Iterable<ScannerReport.ContextProperty> it = from(cache.getAll().entrySet()).transform(new MapEntryToContextPropertyFunction());
-    writer.writeContextProperties(it);
+    MapEntryToContextPropertyFunction transformer = new MapEntryToContextPropertyFunction();
+
+    Iterable<ScannerReport.ContextProperty> fromCache = from(cache.getAll().entrySet())
+      .transform(transformer);
+
+    Iterable<ScannerReport.ContextProperty> fromSettings = from(settings.getProperties().entrySet())
+      .filter(e -> e.getKey().startsWith("sonar.analysis."))
+      .transform(transformer);
+
+    writer.writeContextProperties(Iterables.concat(fromCache, fromSettings));
   }
 
   private static final class MapEntryToContextPropertyFunction implements Function<Map.Entry<String, String>, ScannerReport.ContextProperty> {
     private final ScannerReport.ContextProperty.Builder builder = ScannerReport.ContextProperty.newBuilder();
 
     @Override
-    public ScannerReport.ContextProperty apply(Map.Entry<String, String> input) {
+    public ScannerReport.ContextProperty apply(@Nonnull  Map.Entry<String, String> input) {
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) Remove this "Nonnull" annotation to honor the overridden method's contract. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS2638)